src/sentry/incidents/logic.py

```diff
@@ -52,11 +62,58 @@ def create_incident(
             IncidentGroup.objects.bulk_create([
                 IncidentGroup(incident=incident, group=group) for group in groups
             ])
+
+        if status == IncidentStatus.CREATED:
+            activity_status = IncidentActivityType.CREATED
+        else:
+            activity_status = IncidentActivityType.DETECTED
+
+        create_incident_activity(incident, activity_status, include_event_stats_snapshot=True)
     return incident
 
 
-def build_incident_query_params(incident):
-    params = {'start': incident.date_started, 'end': incident.current_end_date}
+def create_incident_activity(
+    incident,
+    activity_type,
+    value=None,
+    previous_value=None,
+    comment=None,
+    include_event_stats_snapshot=False,
+):
+    snapshot = None
+    if include_event_stats_snapshot:
+        initial_period_length = min(
+            timezone.now() - incident.date_started,
+            MAX_INITIAL_INCIDENT_PERIOD,
+        )
+        end = incident.date_started + initial_period_length
```

Comment:
If the incident runs for more than 7 days should we be snapshotting the most current data? It looks like this will grab up to 7 days after the incident started.

Reply:
For now this snapshot is just intended for the initial creation event, which will never change since it's just intended to show the initial spike. I'll modify this code to be more flexible as part of the actual incident snapshot work.

Reply:
Actually this code is really pretty specific to the creation activity. Maybe I should just create the snapshot in `create_incident` and pass it along to this method.